tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -4028,50 +4033,59 @@ Status ConvertBatchMatMul(OpConverterParams* params) {
   // false}}));
   TF_RETURN_IF_ERROR(
       AllowDataTypes(*params, {DataType::DT_FLOAT, DataType::DT_HALF}));
-  if (inputs[0].is_weights() && inputs[1].is_weights()) {
+  if (inputs.at(0).is_weights() && inputs.at(1).is_weights()) {
     return errors::InvalidArgument(
         "All inputs are weights, but Grappler is expected to fold them.");
   }
+  if (inputs.at(0).is_tensor() && inputs.at(1).is_tensor() &&
+      inputs.at(0).GetTrtDims().nbDims != inputs.at(1).GetTrtDims().nbDims) {
+    return errors::Unimplemented(
+        "Inputs must have the same rank if they are both tensors.");
```

Comment:
I think this is not sufficient. For example, if both are tensors and one with shape [-1, T, C] and the other with shape [-1, C, K], the converter will accept it. But when user provides [N, T, C] and [1, C, K] at runtime the execution will fail. I'm not sure how to fix this, just FYI. May be good to add a note though.

Reply:
Those inputs are valid and will work

Reply:
Does a TRT engine accept inputs with different batch size? I vaguely remember that it will fail enqueue() but I could be wrong.

Reply:
Yes, there is no way to enqueue two inputs with different batch sizes. But you can still have a tensor in the network with TRT dims of [C, K] which is broadcasted over the batch dimension.